### PR TITLE
Rename image to registry.k8s.io/kueue/kueue

### LINKED ISSUE #226: Rename image to registry.k8s.io/kueue/kueue
**What would you like to be cleaned**:

Rename images from `k8s.gcr.io/kueue/kueue` to `registry.k8s.io/kueue/kueue` for the next release.

**Why is this needed**:

https://groups.google.com/a/kubernetes.io/g/dev/c/DYZYNQ_A6_c/m/oD9_Q8Q9AAAJ?utm_medium=email&utm_source=footer

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #226

#### Special notes for your reviewer:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -180,10 +180,9 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
 .PHONY: artifacts
 artifacts: kustomize
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMAGE_TAG}
+	if [ -d artifacts ]; then rm -rf artifacts ; fi
 	@mkdir -p artifacts
-	@rm -r artifacts/*
 	$(KUSTOMIZE) build config/default -o artifacts/manifests.yaml
-	@$(call clean-manifests)
```
Comment (alculquicondor) [inline Makefile:186]: why remove this line?

File: Makefile [hunk 2]
```diff
@@ -180,10 +180,9 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
 .PHONY: artifacts
 artifacts: kustomize
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMAGE_TAG}
+	if [ -d artifacts ]; then rm -rf artifacts ; fi
```
Comment (alculquicondor) [inline Makefile:183]:
```suggestion
	if [ -d artifacts ]; then rm -rf artifacts; fi
```

**Review by alculquicondor — COMMENTED** [review]
Since the staging URL is not changing, there is nothing else to do in the Makefile, right?

File: Makefile (hunk 1 above)
Comment (ArangoGutierrez) [reply Makefile:186]: When cutting a release, we want to `kustomize set image` to stay, not to rewrite it back to `gcr.io` "staging" 
See release-0.1 the config/manager/kustomize goes to staging it should go to k8s.io or after this PR to registry.k8s

Comment (alculquicondor) [reply Makefile:186]: What I did last time is actually update the value in the definition of `clean-manifests`.

But maybe what you propose is good.

**Pull request merged**